src/app/zap-templates/templates/app/helper.js

```diff
@@ -178,6 +180,30 @@ function chip_endpoint_generated_functions()
   return ret.concat('\n');
 }
 
+function chip_endpoint_generated_commands_list()
+{
+  let ret = '{';
+  this.clusterList.forEach((c) => {
+    clientGeneratedCommands = [];
+    serverGeneratedCommands = [];
+    c.commands.forEach((cmd) => {
+      console.log(cmd);
+      if (cmd.mask.includes('incoming_server')) {
+        clientGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+      if (cmd.mask.includes('incoming_client')) {
+        serverGeneratedCommands.push(`${cmd.commandId} /* ${cmd.name} */`);
+      }
+    });
+    clientGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
+    serverGeneratedCommands.push("0xFFFF'FFFF /* end of list */");
```

Comment:
Only needs to be done if the lists are nonempty; see below.  And then we would need to be a bit more careful to not throw in spurious `','` for the empty-list case.

Reply:
Updated.